packages/contracts-bedrock/src/dispute/PermissionedDisputeGame.sol

```diff
@@ -22,6 +22,34 @@ import { IBigStepper } from "src/dispute/interfaces/IBigStepper.sol";
 ///         costs that certain networks may not wish to support. This contract can also be used as a fallback mechanism
 ///         in case of a failure in the permissionless fault proof system in the stage one release.
 contract PermissionedDisputeGame is FaultDisputeGame {
+    /// @custom:field _gameType The type ID of the game.
+    /// @custom:field _absolutePrestate The absolute prestate of the instruction trace.
+    /// @custom:field _maxGameDepth The maximum depth of bisection.
+    /// @custom:field _splitDepth The final depth of the output bisection portion of the game.
+    /// @custom:field _clockExtension The clock extension to perform when the remaining duration is less than the
+    /// extension.
+    /// @custom:field _maxClockDuration The maximum amount of time that may accumulate on a team's chess clock.
+    /// @custom:field _vm An onchain VM that performs single instruction steps on an FPP trace.
+    /// @custom:field _weth WETH contract for holding ETH.
+    /// @custom:field _anchorStateRegistry The contract that stores the anchor state for each game type.
+    /// @custom:field _l2ChainId Chain ID of the L2 network this contract argues about.
+    /// @custom:field _proposer Address that is allowed to create instances of this contract.
+    /// @custom:field _challenger Address that is allowed to challenge instances of this contract.
+    struct PDGConstructorParams {
```

Comment:
Does it still solve stack too deep if we remove proposer and challenger from this struct and have those as individual inputs? The rationale being we can then keep the interface between the PDG and regular FDG more similar, and have them both take the same `FDGConstructorParams` arg, then PDF just takes two extra args

Reply:
@mds1 yeah I was thinking the same, would be nicer that way imo.

Reply:
It compiles with forge coverage this way but would need to be excluded from interfaces check. Couldn't find a way to use this struct for both contracts without breaking the interfaces check for IPermissionedDisputeGame.sol. The only option is verbose and probably unneccesarily more costly gas-wise.

```solidity
constructor (DGConstructorParams memory _params) FaultDisputeGame(FaultDisputeGame.DGConstructorParams({
        _gameType: _params._gameType,
        _absolutePrestate: _params._absolutePrestate,
        ...
})) {
        ...
}
```

Reply:
Hmm, ok, I'll take a look at this and see if I can get it to pass the checks